kotlin_client/src/main/java/com/github/paylike/kotlin_client/dto/HintsResponse.kt

```diff
@@ -0,0 +1,9 @@
+package com.github.paylike.kotlin_client.dto
+
+// Describes the hints array received after executing a challenge successfully.
+class HintsResponse(json: Map<String, Any>){
+    val hints: List<String> = listOf()
+    init {
+        (json["hints"] as List<Any>).forEach { hints.plus(it.toString()) }
```

Comment:
plus returns a list but does not extend your list

`listOf` is immutable
use `mutableListOf` and with that you can use `add()` which will extend your list!